posthog/cdp/templates/microsoft_teams/template_microsoft_teams.py

```diff
@@ -0,0 +1,62 @@
+from posthog.cdp.templates.hog_function_template import HogFunctionTemplate, HogFunctionSubTemplate, SUB_TEMPLATE_COMMON
+
+template: HogFunctionTemplate = HogFunctionTemplate(
+    status="beta",
```

Comment:
Same question here as for discord. Perhaps we want to make this free?

Reply:
Yeah, we should make this free.

I don't seem to find a way to check if a string starts with `https://%.webhook.office.com/webhookb2/` in hog.

I tried this code: 
```
if (not inputs.webhookUrl like 'https://%.webhook.office.com/webhookb2/') {
    throw Error('Invalid url');
}
```

But that will allow this URL `https://webhook.site/571d559e-238d-4305-90c3-d36ec91fe7d5#https://max.webhook.office.com/webhookb2/`

Reply:
You'll need the `%` in the end as well... so `url like 'https://%.webhook.office.com/webhookb2/%'`

Reply:
I tried the syntax from the standup today but it's still possible to trick the check by including the correct URL as a fragment:
![2024-10-18 at 12 03 29@2x](https://github.com/user-attachments/assets/5da675f9-6edf-4f57-bcb1-7a536d2cdfd0)

It doesn't seem like it's possible to filter for `string startsWith 'https://%.webhook.office.com/webhookb2/%'`

Reply:
Right... in this case the first `%` in `https://%.` is effectively matches the string `'webhook.site/57...d5#https://max'`. We'll need better URL parsing to get past this. 

Some ideas:
- Use a regex and be strict about what characters are allowed
- Split the initial URL and discard everything after `?` and `#` symbols
- ClickHouse has a bunch of [URL functions](https://clickhouse.com/docs/en/sql-reference/functions/url-functions) and we could port some of them over.

Reply:
I am going to use this condition:
```
if (not match(inputs.webhookUrl, '^https://[^/]+.webhook.office.com/webhookb2/.*'))
```